### PR TITLE
refactor!: remove interbtc maintenance fund, but transfer leftover to…

### PR DESCRIPTION
**Pull request by gregdhill**
… treasury

Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

To accompany https://github.com/interlay/interbtc-spec/pull/25

### CODE REVIEW

File: crates/fee/src/lib.rs [hunk 1]
```diff
@@ -295,15 +259,10 @@ impl<T: Config> Pallet<T> {
...
-
-        // give remaining rewards to maintainer (dev fund)
-        let maintainer_rewards = amount.saturating_sub(vault_rewards);
-        let maintainer_account_id = Self::maintainer_account_id();
-        ext::treasury::transfer::<T>(&Self::fee_pool_account_id(), &maintainer_account_id, maintainer_rewards)?;
-
+        // distribute vault rewards and return leftover
+        let remaining = Self::distribute::<_, _, T::SignedFixedPoint, T::VaultRewards>(amount)?;
+        // sweep the remaining rewards to the treasury
+        T::OnSweep::on_sweep(&Self::fee_pool_account_id(), remaining)?;
```
Comment (sander2) [inline crates/fee/src/lib.rs:265]: Should we add a check such we don't call this if remaining is zero?

File: crates/reward/src/lib.rs [hunk 2]
```diff
@@ -295,8 +295,8 @@ pub trait Rewards<AccountId> {
     /// Deposit an `amount` of stake to the `account_id`.
     fn deposit_stake(account_id: &AccountId, amount: Self::SignedFixedPoint) -> Result<(), DispatchError>;
 
-    /// Distribute the `reward` to all participants.
-    fn distribute(reward: Self::SignedFixedPoint) -> Result<Self::SignedFixedPoint, DispatchError>;
+    /// Distribute the `reward` to all participants OR return the leftover.
```
Comment (sander2) [inline crates/reward/src/lib.rs:298]: needs spec update

**Comment by sander2** [comment]
I am not convinced it's a good idea to not give maintainers a part of the fees. But IF that has been decided, then if you address the comments, this looks good

**Comment by gregdhill** [comment]
> I am not convinced it's a good idea to not give maintainers a part of the fees.

There will soon be another source of income for the maintainers (INTR) and if we also remove redeem fees it would make sense that all (interBTC) rewards are given to Vaults.

**Review by sander2 — APPROVED** [review]

**Pull request merged**